### DISCUSSION TITLE
Querying hierarchies?

### DISCUSSION THREAD

**Discussion by Metadorius** [bevyengine/bevy#5498] [Q&A] [answered]
What's the most optimized way to query for a certain hierarchy of items? I am implementing an RTS camera that I need to orbit around a pivot which will be moving. I skimmed through the docs and seen only a possibility to query for entities with children/a parent, without specifying what components does the parent/the children have.

**Comment by rparrett** [comment, marked as answer]
One thing that is possible now in Bevy 0.8 is iterating over `Children` with `iter_many`.

```rust
fn children(query: Query<&Children>, text_query: Query<&Text>) {
    for children in query.iter() {
        for child in text_query.iter_many(children) {
            info!("{}", child.sections[0].value);
        }
    }
}
```

But other than that, we pretty much have to do a lot of manual work with multiple queries, e.g. 

```rust
fn parent(query: Query<&Parent>, text_query: Query<&Text>) {
    for parent in query.iter() {
        for text in text_query.get(parent.get()) {
            info!("{}", text.sections[0].value);
        }
    }
}
```

It's also possible to construct recursive queries like this: https://github.com/bevyengine/bevy/issues/838#issuecomment-772082427 (updating visibility hierarchies this way is thankfully no longer necessary with Bevy 0.8)

There's ongoing discussion about entity-entity relations here:
https://github.com/bevyengine/bevy/issues/3742
